sdk/src/stream_builder/build/build_stream_topic.rs

```diff
@@ -0,0 +1,98 @@
+use crate::client::{StreamClient, TopicClient};
+use crate::clients::client::IggyClient;
+use crate::compression::compression_algorithm::CompressionAlgorithm;
+use crate::error::IggyError;
+use crate::identifier::{IdKind, Identifier};
+use crate::stream_config::IggyConsumerConfig;
+use crate::utils::expiry::IggyExpiry;
+use crate::utils::topic_size::MaxTopicSize;
+use tracing::info;
+
+/// Builds an `IggyStream` and `IggyTopic` if any of them does not exists
+/// and if the boolean flags to create them are set to true. In that case it will build
+/// them using the given `IggyClient` and `IggyProducerConfig`.
+///
+/// If the boolean flags to create them are set to false, it will not build them and will return
+/// and return Ok(()) since this is expected behavior.d
+///
+/// # Arguments
+///
+/// * `client` - The `IggyClient` to use.
+/// * `config` - The `IggyProducerConfig` to use.
+///
+/// # Errors
+///
+/// * `IggyError` - If the iggy stream topic cannot be build.
+///
+pub(crate) async fn build_iggy_stream_topic_if_not_exists(
+    client: &IggyClient,
+    config: &IggyConsumerConfig,
+) -> Result<(), IggyError> {
+    let stream_id = config.stream_id();
+    let stream_name = config.stream_name();
+    let topic_id = config.topic_id();
+    let topic_name = config.topic_name();
+
+    info!("Check if stream exists.");
```

Comment:
As mentioned previously, let's keep most of these comments at trace/debug level :)